src/psyclone/tests/psyir/symbols/symbol_table_test.py

```diff
@@ -1546,31 +1646,37 @@ def test_deep_copy():
     new identical copies of the symbols in the original symbol table'''
 
     # Create an initial SymbolTable
-    symtab = SymbolTable(default_visibility=Symbol.Visibility.PRIVATE)
+    symtab = symbols.SymbolTable(
+        default_visibility=symbols.Symbol.Visibility.PRIVATE)
     dummy = Schedule(symbol_table=symtab)
-    mod = ContainerSymbol("my_mod")
-    sym1 = DataSymbol("symbol1", INTEGER_TYPE,
-                      interface=ArgumentInterface(
-                          ArgumentInterface.Access.READ))
-    sym2 = Symbol("symbol2", interface=ImportInterface(mod,
-                                                       orig_name="altsym2"))
-    sym3 = DataSymbol("symbol3", INTEGER_TYPE)
+    mod = symbols.ContainerSymbol("my_mod")
+    sym1 = symbols.DataSymbol("symbol1", symbols.INTEGER_TYPE,
+                              interface=symbols.ArgumentInterface(
+                                  symbols.ArgumentInterface.Access.READ))
+    sym2 = symbols.Symbol(
+        "symbol2",
+        interface=symbols.ImportInterface(mod, orig_name="altsym2"))
+    sym3 = symbols.DataSymbol("symbol3", symbols.INTEGER_TYPE)
     symtab.add(mod)
     symtab.add(sym1)
     symtab.add(sym2, tag="tag1")
     symtab.add(sym3)
     symtab.specify_argument_list([sym1])
+    rsym = symbols.RoutineSymbol("my_sub")
+    gisym = symbols.GenericInterfaceSymbol("generic_sub", [(rsym, False)])
+    symtab.add(rsym)
+    symtab.add(gisym)
```

Comment:
Unless I am missing it, I think we still need to check that this symbols are not part of the GenericInterface after the copy and that the internal links have been updated (similar to what we do with the ImportInterface)

Reply:
Oops, you're right - I'd missed that. Got too excited by the fact it didn't crash anymore :-)